pom.xml

```diff
@@ -41,28 +43,23 @@
 			<artifactId>javamelody-core</artifactId>
 			<version>1.94.0</version>
 		</dependency>
-		<!-- slf4j-api for Jenkins issue https://issues.jenkins-ci.org/browse/JENKINS-11293 
-			NoClassDefFoundError: org.slf4j.ILoggerFactory on IBM J9 JVM, (this is not 
-			be needed anymore since with JENKINS-12650, slf4j-api and slf4j-jdk14 are 
-			now bundled in core), removed for https://issues.jenkins-ci.org/browse/JENKINS-46966 
-			<dependency> <groupId>org.slf4j</groupId> <artifactId>slf4j-api</artifactId> 
-			<version>1.6.3</version> </dependency> -->
 		<dependency>
 			<groupId>com.lowagie</groupId>
 			<artifactId>itext</artifactId>
 			<version>2.1.7</version>
 			<exclusions>
+				<!-- Provided by bouncycastle-api plugin -->
```

Comment:
it it excluded here, why should it be provided ?

Reply:
Thank Just to keep the dependency tree more pure. iText claims to depend on BouncyCastle, so if we can dynamically link against these classes via a plugin-to-plugin dependency on the Jenkins `bouncycastle-api` plugin, then why not. Excluding the classes without providing them via alternate means could possibly be safe, as long as we never invoke any code paths that consume this API. Safer to dynamically link against these classes in case they are used by some codepath.

Reply:
yes, it is safe to exclude

Reply:
I see no evidence in support of the claim that it is safe to exclude this dependency from the classpath. I have provided evidence against this claim; namely, that the iText POM claims to depend on BouncyCastle.

Reply:
I know iText 2.1.7 and what the plugin and javamelody-core I have written do with it. I claim that it is safe to exclude bouncycastle.
The plugin or javamelody-core use iText like this in thousands of servers, so it is a safe claim.

Reply:
By the way, the plugin has excluded bouncycastle for many years and there was no issue declared for that.

Reply:
> I know iText 2.1.7 and what the plugin and javamelody-core I have written do with it. I claim that it is safe to exclude bouncycastle.

That is an [argument from authority](https://en.wikipedia.org/wiki/Argument_from_authority), which is a logical fallacy.

Reply:
=> commit e299c74fdef19af0a0cc3f334fffd8a85349c004